src/elements/AnnotationBlock.ts

```diff
@@ -0,0 +1,142 @@
+import { Annotation } from "../types/Annotation";
+import { CancelButton } from "./CancelButton";
+import { DeleteButton } from "./DeleteButton";
+import { SaveButton } from "./SaveButton";
+
+/**
+ * HTML div element associated with an annotation, which can be made editable to udpate
+ * or delete its associated annotation.
+ */
+class AnnotationBlock extends HTMLDivElement {
+    annotation: Annotation;
```

Comment:
I'm wondering if the cached annotation here is the cause of the problem you noticed with saving changes to the selection. It seems like it would be better to use the annotorious method to get annotation by id when needed.

I've experimented with it some and I think the changed selection target isn't getting stored. There's a change selection target event we could listen to, but I'm not sure that we should (it fires repeatedly when dragging to adjust).
https://recogito.github.io/annotorious/api-docs/osd-plugin/#changeselectiontarget

Reply:
That's helpful, thanks! I agree, let's file it as an issue (maybe on the geniza repo to keep things tidy) and merge this in without fixing.